core/src/inputs/advice/mod.rs

```diff
@@ -114,4 +122,25 @@ impl AdviceSet {
             Self::MerklePathSet(set) => set.update_leaf(index, value),
         }
     }
+
+    /// Adds the specified Merkle path to the Merkle path set. The `index` and `value` parameters
+    /// specify the leaf node at which the path starts.
+    ///
+    /// # Errors
+    /// Returns an error if:
+    /// - The specified index is not valid in the context of this Merkle path set (i.e., the index
+    ///   implies a greater depth than is specified for this set).
+    /// - The specified path is not consistent with other paths in the set (i.e., resolves to a
+    ///   different root).
+    pub fn add_path(
+        &mut self,
+        index: u64,
+        value: Word,
+        path: Vec<Word>,
+    ) -> Result<(), AdviceSetError> {
+        match self {
+            Self::MerklePathSet(set) => set.add_path(index, value, path),
+            _ => Ok(()),
+        }
+    }
```

Comment:
This method is relevant only for `MerklePathSet` variant, and I'm not sure it is a good idea to expose it on the overall `AdviceSet`. I think a better way would be to pass all required info to the `new_merkle_path_set()` constructor so that it builds the `MerklePathSet` at construction time.

Reply:
Is `new_merkle_path_set()` an existing constructor or is it to be implemented?

Reply:
It could be the constructor you've defined on line 53 - it would just look different (i.e., would take more input parameters and would call `set.add_path()` internally).